tests/python_functional/functional_tests/secure-logging/test_secure_logging.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import os
+
+import pytest
+
+from src.helpers.secure_logging.conftest import *  # noqa:F403, F401
+
+example_message = "example-message"
+num_of_messages = 3
+
+
+@pytest.mark.skipif("cmake_build" in os.environ, reason="only autotools is supported for now")
+def test_secure_logging(config, syslog_ng, secure_logging):
+    generator_source = config.create_example_msg_generator_source(num=num_of_messages, template=config.stringify(example_message))
+    template = "$(slog -k {} -m {} $MSG)".format(secure_logging["derived_key"], secure_logging["cmac"])
+    file_destination = config.create_file_destination(file_name="output.log", template=config.stringify(template + '\n'))
+
+    config.create_logpath(statements=[generator_source, file_destination])
+    syslog_ng.start(config)
```

Comment:
While syntax checking syslog-ng config I have got the following WARNINGS in console log, could we fix them?:
```
[2020-04-01T06:33:10.128794] [SLOG] WARNING: MAC file not found; File='/source/tests/python_functional/reports/2020-04-01-06-33-10-002621/test_secure_logging/cmac'
[2020-04-01T06:33:10.128820] [SLOG] WARNING: Aggregated MAC not found or invalid; File='/source/tests/python_functional/reports/2020-04-01-06-33-10-002621/test_secure_logging/cmac'
```

Reply:
Unfortunaly, this is a known issue. Even though it is fine that cmac does not exist (it will be created by syslog-ng during the first evaluation), it emits this warning.

I would like to avoid changing the c code at this point. This will be fixed as part of a larger work.

Reply:
Ok, thank you for the explanation. It is ok, if it is known issue. I have just thought we can handle this from test environment.